File: Day247/equilibriumPoint.java
package Day247;

public class equilibriumPoint {

    public static int equiPoint(int arr[], int n) {

        int ts = 0;
        for (int i=0; i<n; i++) {
            ts += arr[i];
        }
        int p = 0;
        for (int i=0; i<n; i++) {
            System.out.println(p + " " + (ts - (p+arr[i])));
            if (p == (ts - (p+arr[i]))) {
                return i+1;
            } else {
                p += arr[i];
            }

        }
        return -1;
    }

    public static void main(String[] args) {
        int arr[] = {1,3,5,2,2};
        int n = 5;
        System.out.println(equiPoint(arr, n));
    }
}

File: Day247/equilibriumPoint.go
package main 

import (
	"fmt"
)

func equilibriumPoint(arr []int, n int) int {
	ts, p := 0, 0 
	for _,v := range arr {
		ts += v
	}
	for i, v := range arr{
		if p == ts - p - v {
			return i+1
		} else {
			p += v
		}
	}
	return -1
}

func main() {
	n := 5
	arr := []int{1,3,5,2,2}
	fmt.Println(equilibriumPoint(arr, n))
}